src/components/CheckboxRadio/CheckboxRadio.vue

```diff
@@ -0,0 +1,347 @@
+		<br>
+		sharingEnabled: {{ sharingEnabled }}
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			sharingEnabled: false,
+		}
+	}
+}
+</script>
+```
+
+### Standard radio set
+```vue
+<template>
+	<div>
+		<CheckboxRadio :checked.sync="sharingPermission" value="r" name="sharing_permission_radio" type="radio">Default permission read</CheckboxRadio>
+		<CheckboxRadio :checked.sync="sharingPermission" value="rw" name="sharing_permission_radio" type="radio">Default permission read+write</CheckboxRadio>
+		<br>
+		sharingPermission: {{ sharingPermission }}
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			sharingPermission: 'r',
+		}
+	}
+}
+</script>
+```
+
+### Standard checkbox set
+```vue
+<template>
+	<div>
+		<CheckboxRadio :disabled="true" :checked.sync="sharingPermission" value="r" name="sharing_permission">Permission read</CheckboxRadio>
+		<CheckboxRadio :checked.sync="sharingPermission" value="w" name="sharing_permission">Permission write</CheckboxRadio>
+		<CheckboxRadio :checked.sync="sharingPermission" value="d" name="sharing_permission">Permission delete</CheckboxRadio>
+		<br>
+		sharingPermission: {{ sharingPermission }}
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			sharingPermission: ['r', 'd'],
+		}
+	}
+}
+</script>
+```
+
+</docs>
+
+<template>
+	<element :is="wrapperElement"
+		:class="{
+			'checkbox-radio--checked': isChecked,
+			'checkbox-radio--disabled': disabled,
+			'checkbox-radio--indeterminate': indeterminate,
+		}"
+		class="checkbox-radio">
+		<input :id="id"
+			:checked="isChecked"
+			:disabled="disabled"
+			:indeterminate="indeterminate"
+			:name="name"
+			:type="type"
+			:value="value"
+			class="checkbox-radio__input"
+			@change="onToggle">
+
+		<label :for="id" class="checkbox-radio__label">
+			<icon :is="checkboxRadioIconElement"
+				:size="24"
+				class="checkbox-radio__icon"
+				title=""
+				decorative />
+
+			<!-- @slot The checkbox/radio label -->
+			<slot />
+		</label>
+	</element>
+</template>
+
+<script>
+import CheckboxBlankOutline from 'vue-material-design-icons/CheckboxBlankOutline'
+import MinusBox from 'vue-material-design-icons/MinusBox'
+import CheckboxMarked from 'vue-material-design-icons/CheckboxMarked'
+import RadioboxMarked from 'vue-material-design-icons/RadioboxMarked'
+import RadioboxBlank from 'vue-material-design-icons/RadioboxBlank'
+
+import GenRandomId from '../../utils/GenRandomId'
+import l10n from '../../mixins/l10n'
+
+export const TYPE_CHECKBOX = 'checkbox'
+export const TYPE_RADIO = 'radio'
+
+export default {
+	name: 'CheckboxRadio',
+
+	mixins: [l10n],
+
+	props: {
+
+		/**
+		 * Unique id attribute of the input
+		 */
+		id: {
+			type: String,
+			default: () => 'checkbox-radio-' + GenRandomId(),
+			validator: id => id.trim() !== '',
+		},
+
+		/**
+		 * Input name. Especially required for radios set
```

Comment:
"especially required" -> is it required or not?

Reply:
You can't do conditional required.
It is for radio, unless you want a single radio, which you can, buuuuut doesn't make any ses=nse xD

Reply:
then say "required for radio, optional for checkbox"?

Reply:
:+1: